solidity/interfaces/middleware/IInterchainAccountRouter.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.6.11;
+
+import {OwnableMulticall} from "../../contracts/OwnableMulticall.sol";
+import {CallLib} from "../../contracts/libs/Call.sol";
+
+interface IInterchainAccountRouter {
+    function callRemote(
+        uint32 _destinationDomain,
+        CallLib.Call[] calldata calls
+    ) external returns (bytes32);
+
+    function callRemoteWithOverrides(
+        uint32 _destinationDomain,
+        bytes32 _router,
+        bytes32 _ism,
+        CallLib.Call[] calldata calls
+    ) external returns (bytes32);
+
+    function getLocalInterchainAccount(
+        uint32 _origin,
+        bytes32 _router,
+        bytes32 _sender,
+        address _ism
+    ) external view returns (OwnableMulticall);
+
+    function getLocalInterchainAccount(
+        uint32 _origin,
+        address _router,
+        address _sender,
+        address _ism
+    ) external view returns (OwnableMulticall);
```

Comment:
not sure I agree with the name changes here
at some point we should really stabilize the interfaces

Reply:
My goal here was to clarify that the function previously known as `getInterchainAccount` could only be used to get the address for contracts on the local chain. Maybe that was clear already, I'm happy to revert.

I do have a slight preference for `callRemote` over `dispatch`, but I don't feel strongly and am happy to discuss

Reply:
happy with `callRemote`
can `getInterchainAccount` be used for nonlocal defaults?

Reply:
Only for EVM chains since it uses CREATE2 for address derivation.